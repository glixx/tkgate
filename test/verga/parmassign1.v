module top;
   wire w1,w2;
   reg p,q;

   foo f1 (.z(w1),.b(p),.a(q));
   foo f2 (.z(w2),.b(p),.a(q));

   initial $monitor("w1=%b  w2=%b  p=%b  q=%b  f1.v=%b  f2.v=%b",w1,w2,p,q,top.f1.v,top.f2.v);

   initial $stop(7);

   initial
     begin   
	#1 p = 1'b0; q = 1'b0;
	#1 p = 1'b0; q = 1'b1;
	#1 p = 1'b1; q = 1'b0;
	#1 p = 1'b1; q = 1'b1;
	$display("changeing f1.v");
	top.f1.v = 1'b1;
	#1 p = 1'b0; q = 1'b0;
	#1 p = 1'b0; q = 1'b1;
	#1 p = 1'b1; q = 1'b0;
	#1 p = 1'b1; q = 1'b1;
	#1;
     end   
endmodule

module foo(z,a,b);
output z;
input a;
input b;
reg v;

  initial v = 0;

  assign z = a ^ b ^ v;

endmodule
