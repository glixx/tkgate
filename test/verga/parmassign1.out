w1=x  w2=x  p=x  q=x  f1.v=0  f2.v=0
w1=0  w2=0  p=0  q=0  f1.v=0  f2.v=0
w1=1  w2=1  p=0  q=1  f1.v=0  f2.v=0
w1=1  w2=1  p=1  q=0  f1.v=0  f2.v=0
changeing f1.v
w1=1  w2=0  p=1  q=1  f1.v=1  f2.v=0
w1=1  w2=0  p=0  q=0  f1.v=1  f2.v=0
w1=0  w2=1  p=0  q=1  f1.v=1  f2.v=0
